actions/mmdebstrap_action.go

```diff
@@ -0,0 +1,193 @@
+ If no components are specified debos will use main as default.
+
+Example:
+ components: [ main, contrib ]
+
+- keyring-packages -- list of keyrings for package validation.
+
+- keyring-files -- list keyring files for repository validation.
+
+- merged-usr -- use merged '/usr' filesystem, true by default.
+
+- include -- list of packages to install during bootstrap.
+
+*/
+package actions
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"runtime"
+
+	"github.com/go-debos/debos"
+	"github.com/go-debos/fakemachine"
+)
+
+type MmdebstrapAction struct {
+	debos.BaseAction `yaml:",inline"`
+	Suite            string
+	Mirrors          []string
+	Variant          string
+	KeyringPackages  []string `yaml:"keyring-packages"`
+	KeyringFiles     []string `yaml:"keyring-files"`
+	Components       []string
+	MergedUsr        *bool `yaml:"merged-usr"`
+	Include          []string
+}
+
+func NewMmdebstrapAction() *MmdebstrapAction {
+	d := MmdebstrapAction{}
+	// Be secure by default
+	// Use main as default component
+	d.Components = []string{"main"}
+
+	return &d
+}
+
+func (d *MmdebstrapAction) listOptionFiles(context *debos.DebosContext) []string {
+	files := []string{}
+
+	if d.KeyringFiles != nil {
+		for _, file := range d.KeyringFiles {
+			file = debos.CleanPathAt(file, context.RecipeDir)
+			files = append(files, file)
+		}
+	}
+
+	return files
+}
+
+func (d *MmdebstrapAction) Verify(context *debos.DebosContext) error {
+	if len(d.Suite) == 0 {
+		return fmt.Errorf("suite property not specified")
+	}
+
+	files := d.listOptionFiles(context)
+
+	// Check if all needed files exists
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *MmdebstrapAction) PreMachine(context *debos.DebosContext, m *fakemachine.Machine, args *[]string) error {
+
+	mounts := d.listOptionFiles(context)
+
+	// Mount configuration files outside of recipes directory
+	for _, mount := range mounts {
+		m.AddVolume(path.Dir(mount))
+	}
+
+	return nil
+}
+
+func (d *MmdebstrapAction) Run(context *debos.DebosContext) error {
+	cmdline := []string{"mmdebstrap"}
+
+	if d.MergedUsr != nil {
+		if *d.MergedUsr {
+			cmdline = append(cmdline, "--include=usrmerge")
+		} else {
+			cmdline = append(cmdline, "--hook-dir=/usr/share/mmdebstrap/hooks/no-merged-usr")
+		}
+	}
+
+	if d.KeyringFiles != nil {
+		s := strings.Join(d.KeyringFiles, ",")
+		cmdline = append(cmdline, fmt.Sprintf("--keyring=%s", s))
+	}
+
+	if d.KeyringPackages != nil {
+		s := strings.Join(d.KeyringPackages, ",")
+		cmdline = append(cmdline, fmt.Sprintf("--include=%s", s))
+	}
+
+	if d.Components != nil {
+		s := strings.Join(d.Components, ",")
+		cmdline = append(cmdline, fmt.Sprintf("--components=%s", s))
+	}
+
+	/* Only works for amd64, arm64 and riscv64 hosts, which should be enough */
+	foreign := context.Architecture != runtime.GOARCH
+
+	if foreign {
+		cmdline = append(cmdline, fmt.Sprintf("--architectures=%s", context.Architecture))
```

Comment:
with mmdebstrap you can avoid this hack; just always set the `--architecture` option

Reply:
Fixed in latest force push